src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy

```diff
@@ -2,23 +2,34 @@ package com.avioconsulting.mule.linter.model
 
 class Application {
 
+    static final String APPLICATION_DOES_NOT_EXIST = 'Application directory does not exists.'
+    static final String POM_FILE = 'pom.xml'
+
     File applicationPath
+    List<ProjectFile> files = []
 
     Application(File applicationPath) {
         this.applicationPath = applicationPath
+        if (!this.applicationPath.exists()) {
+            throw new FileNotFoundException(applicationPath.absolutePath, APPLICATION_DOES_NOT_EXIST)
+        }
     }
 
-    Boolean hasFile(String filename) {
-        File file = new File(applicationPath, filename)
-        if(!file.exists() || !file.canRead() || file.length() == 0) {
-            return false
-        }
+    File getApplicationPath() {
+        return applicationPath
+    }
 
-        return true
+    void setApplicationPath(File applicationPath) {
+        this.applicationPath = applicationPath
+    }
+
+    PomFile getPomFile() {
```

Comment:
We said this should be created as a private variable in the constructor

Reply:
added private variable and returning that now.